packages/checkbox/src/vaadin-checkbox.js

```diff
@@ -184,6 +184,21 @@ class Checkbox extends SlotLabelMixin(
     return this.$.noop;
   }
 
+  /**
+   * Override __copyNodesToSlotTarget from SlotTargetMixin to show a warning.
+   * @override
+   * @protected
+   * @param {Array<Node>} nodes
+   **/
+  __copyNodesToSlotTarget(nodes) {
+    super.__copyNodesToSlotTarget(nodes);
+
+    console.warn(
+      `WARNING: Since Vaadin 22, placing the label as a direct child of a <vaadin-checkbox> is deprecated.
+Please use <label slot="label"> wrapper or the "label"-property instead.`
+    );
```

Comment:
BTW maybe we could have an utility in `component-base` e.g. something like this:

```js
export function warnDeprecated(version, feature, replacement) {
    console.warn(
      `WARNING: Since Vaadin ${version}, "${feature} is deprecated. Please use ${replacement} instead.`
    );
}
```

Then we can use it like this:

```js
warnDeprecated(
  '22', 
  'placing the label as a direct child of a <vaadin-checkbox>',
  '<label slot="label"> wrapper or the "label"-property'
)
```

Apart from readability, this will make it really easy to find all the deprecation warnings.